test/web-platform-tests/utils.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+const { Canvas } = require("../../lib/jsdom/utils.js");
+
+const nodeMajor = Number(process.versions.node.split(".", 1)[0]);
+const hasNode10 = nodeMajor >= 10;
+const hasNode11 = nodeMajor >= 11;
+const hasNode12 = nodeMajor >= 12;
+const hasCanvas = Boolean(Canvas);
+
+exports.resolveReason = reason => {
+  return {
+    shouldSkip:
```

Comment:
I like this function. But I think it reveals a chance for improvement. Since shouldSkip and expectFail are mutually exclusive, instead it can return one of "run", "skip", "expect-fail".